### PR TITLE
Manage authn/authz/groupfinder from settings (fixes #40)

### LINKED ISSUE #40: Manage authentication from settings
Currently, authentication method is setup with code in ``cliquet.initialize()``. [See docs](http://cliquet.readthedocs.org/en/latest/configuration.html#custom-authentication)

They could be specified from configuration files, just like storage backend: 

```
cliquet.authentication_policies  = storage.authentication.basic_auth,
                                   oauth2.providers.google
```

And we could mimic the ``load_from_config()`` to instantiate the class from potential settings

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by leplatrem** [comment]
Tests fail because it requires last master of pyramid_multiauth with this https://github.com/mozilla-services/pyramid_multiauth/pull/6

**Comment by Natim** [comment]
We should change "fxa_uid" to "fxa:uid" same for other policies.

File: cliquet/__init__.py [hunk 1]
```diff
@@ -76,6 +75,13 @@
     'fxa-oauth.scope': 'profile',
     'fxa-oauth.state.ttl_seconds': 3600,  # 1 hour
     'fxa-oauth.webapp.authorized_domains': '',
+    'multiauth.policies': 'fxa',
+    'multiauth.policy.fxa.use': ('cliquet.authentication.'
```
Comment (ametaireau) [inline cliquet/__init__.py]: I like the ".use".

File: cliquet/initialization.py [hunk 2]
```diff
@@ -73,25 +73,22 @@ def _redirect_to_version_view(request):
...
+    # Track policy for logging.
+    def on_policy_selected(event):
+        value = event.policy.__class__.__name__
+        value = value.replace('Authentication', '').replace('Policy', '')
+        event.request.authn_type = value
 
-    config.set_authorization_policy(authz_policy)
-    config.set_authentication_policy(authn_policy)
-    config.set_default_permission('readwrite')
+    config.add_subscriber(on_policy_selected, MultiAuthPolicySelected)
```
Comment (ametaireau) [inline cliquet/initialization.py]: Nice, I like this way of handling things :)

File: docs/quickstart.rst [hunk 3]
```diff
@@ -98,6 +99,23 @@ But most likely, it will consist in **defining REST resources** using *Cliquet*
 python API !
 
 
+Authentication
+--------------
+
+Currently, if no authentication is set in settings, *Cliquet* relies on
+:term:`Firefox Accounts`, whose setup is tedious.
```
Comment (ametaireau) [inline docs/quickstart.rst]: I thought this should be "which" but [apparently whose is the right word to use here](http://www.quickanddirtytips.com/education/grammar/whose-for-inanimate-objects?page=all)

File: docs/reference/configuration.rst [hunk 4]
```diff
@@ -222,56 +222,60 @@ in configuration:
     # cliquet.userid_hmac_secret = b4c96a8692291d88fe5a97dd91846eb4
 
 
-Basic Auth
-::::::::::
+Authentication setup
+::::::::::::::::::::
+
+*Cliquet* relies on :github:`pyramid multiauth <mozilla-service/pyramid_multiauth>`_
```
Comment (ametaireau) [inline docs/reference/configuration.rst]: didn't know about this github rule in sphinx!

**Pull request merged**